stock_move_source_relocate/models/stock_move.py

```diff
@@ -18,22 +22,33 @@ def _action_assign(self):
         unconfirmed_moves = unconfirmed_moves.filtered(
             lambda m: m.state in ["confirmed", "partially_available"]
         )
-        unconfirmed_moves._apply_source_relocate()
+        if unconfirmed_moves:
+            unconfirmed_moves._apply_source_relocate()
```

Comment:
Does it change something to check the recordset here?

Should we instead do this check in the `_apply_source_relocate` method, if `self` is empty return.

Reply:
Well, here except the debug log the method will do nothing. And as the Odoo API tends to be "functional-programming" in some way it's OK to call methods on an object without thinking if it's empty or not.